### DISCUSSION TITLE
[Feature] cluster.open需要返回绑定的真实端口

### DISCUSSION THREAD

**Discussion by zaxbbun** [cloudwu/skynet#1643] [Q&A] [answered]
```lua
cluster.open(port)
```
需要返回绑定的真实端口(当port==0时，为系统随机端口），以方便注册服务。
socket.listen已经支持返回端口，但是gateserver部分使用的是socketdriver，不确定怎么修改
@cloudwu

**Comment by cloudwu** [comment, marked as answer]
https://github.com/cloudwu/skynet/commit/b61daaccf1c98d3ca878ca94e9a441d2140e045d  现在让 handler.open 回调函数传回真实的 addr,port 。目前 watchdog 已经可以返回它们了。
https://github.com/cloudwu/skynet/commit/2d5e4ae528aa12d28a9278b02634fa40404aa357  cluster.open 也返回 addr, port

**Comment by cloudwu** [comment]
我觉得 cluster 的配置表中配置 port 0 是没有意义的，所以现在只有 cluster.open(0) 才会返回真实的 port 。这个可能可以用于动态配置。
但还是需要收集所有节点的端口，汇总后再通过 cluster.reload 来组网。

**Reply by zaxbbun** [reply]
我们没有使用clustername组网 使用的服务发现 在线扩展，多版本发布方便一点
